docker/optimism/optimism.sh

```diff
@@ -12,72 +11,102 @@ CACHE_DIR=$(obtain_cache_dir $0)
 DEPLOYMENT_DIR="${CACHE_DIR}/deployment"
 DEPLOYMENT_CONFIG_FILE="${CACHE_DIR}/optimism-local.json"
 KEYFILE="${CACHE_DIR}/${ADDRESS}.json"
+OPTIMISM="${CACHE_DIR}/optimism"
 
 ensure_keyfile_exists ${PRIVKEY} ${KEYFILE}
 echo Beamer deployer\'s keyfile: ${KEYFILE}
 
+e2e_test_op_proof(){
+    local root=$(get_root_dir)
+    local l1_rpc=$1
+    local l2_rpc=$2
+    local privkey=$4
+    local txhash=$5
+    local relayer=${root}/relayer/relayer-node18-linux-x64
+
+    echo Starting OP relayer message prover...
+    timeout 5m bash -c "until ${relayer} prove-op-message \
+                                         --l1-rpc-url '$l1_rpc' \
+                                         --l2-rpc-url '$l2_rpc' \
+                                         --wallet-private-key '$privkey' \
+                                         --l2-transaction-hash '$txhash'; \
+    do sleep 1s; done"
+}
+
+configure_repo() {
+    repo="https://github.com/ethereum-optimism/optimism.git"
+    commit_id="6a474e36aba94f15ed90f71d1920e4b1d34513ab"
+    git clone --no-checkout ${repo} ${OPTIMISM}
+    cd ${OPTIMISM}
+    git checkout ${commit_id}
```

Comment:
May I suggest the following, to avoid cloning the entire history just to get a specific tree?
```
    repo="https://github.com/ethereum-optimism/optimism.git"
    commit_id="6a474e36aba94f15ed90f71d1920e4b1d34513ab"
    git init ${OPTIMISM}
    cd ${OPTIMISM}
    git fetch --depth 1 ${repo} ${commit_id}
    git checkout ${commit_id}
```